xterm-color.el

```diff
@@ -505,11 +546,19 @@ in LIFO order."
                                     xterm-color--current-bg
                                     (/= xterm-color--attributes 0)))
           (has? (attrib)       `(/= (logand ,attrib xterm-color--attributes) 0))
-          (face-cache-get ()   `(gethash (logior (ash xterm-color--attributes 16)
-                                                 (ash (or xterm-color--current-bg 0) 8)
-                                                 (or xterm-color--current-fg 0))
+          (face-cache-get ()   `(gethash (let ((fg (or xterm-color--current-fg 0))
+                                               (bg (or xterm-color--current-bg 0)))
+                                           (if (has? +truecolor+)
+                                               (and xterm-color--cache-truecolor-faces
+                                                 (logior (ash xterm-color--attributes 48)
+                                                         (ash bg 24)
+                                                         fg))
+                                             (logior (ash xterm-color--attributes 16)
+                                                     (ash bg 8)
+                                                     fg)))
```

Comment:
Sorry, this isn't going to behave correctly on 32bit either. I'm going to push a further commit.